src/github.js

```diff
@@ -53,6 +53,22 @@ module.exports = function githubFactory(token, repository) {
      */
     return {
 
+        /**
+         * Verify the credentials for said repository by checking it's info
+         * @returns {Promise<Object>} resolves with the repository data, if the credentials are valid
+         */
+        verifyRepository() {
+            return new Promise((resolve, reject) => {
+                ghrepo.info((err, data) => {
```

Comment:
This library command does indeed require a valid credential to return data, however it doesn't check if the token has the `repo` rights it will need later. (I created a token with fewer rights and the process did fail later).

On one hand, this is an improvement, since it will highlight the error for an expired token. On the other hand, it would be nice if there was a more accurate check. I checked [octonode](https://www.npmjs.com/package/octonode) and didn't see anything obvious.

Do you see any solution?

Reply:
Sorry for the delay in the answer, got stuck with a bunch of stuff.

Soooo, I actually got a few information regarding the repo rights. It seems the API won't provide any information about that whatsoever, BUT, there is a trick that could work if we ever wanted to do this in the future.

Here we have to rely on the GitHub documentation rather than octonode: https://docs.github.com/en/rest/pulls/pulls#create-a-pull-request

A request to create a PR, provided it has the correct information, will return, respectively:
401, if the credentials are wrong;
404, if the credentials lack access and
422 if some information are missing for creating a specific pull request

If we think that would be deeply necessary we could probably "trick" the request by identifying what's the response for said "attempt" to create a PR and, judging by the code of the response we could determine if the credentials are invalid of if the person just don't have access to it, but it's risky.

Anyhow, I didn't think about this until this weekend. Thanks for testing!

I'll merge the code then, since I got your approval.